x-pack/plugins/observability/public/context/has_data_context.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { uniqueId } from 'lodash';
+import React, { createContext, useEffect, useState } from 'react';
+import { Alert } from '../../../alerts/common';
+import { getDataHandler } from '../data_handler';
+import { FETCH_STATUS } from '../hooks/use_fetcher';
+import { usePluginContext } from '../hooks/use_plugin_context';
+import { useTimeRange } from '../hooks/use_time_range';
+import { getObservabilityAlerts } from '../services/get_observability_alerts';
+import { ObservabilityFetchDataPlugins, UXHasDataResponse } from '../typings/fetch_overview_data';
+
+type DataContextApps = ObservabilityFetchDataPlugins | 'alert';
+
+export type HasDataMap = Record<
+  DataContextApps,
+  { status: FETCH_STATUS; hasData?: boolean | UXHasDataResponse | Alert[] }
+>;
+
+export interface HasDataContextValue {
+  hasData: Partial<HasDataMap>;
+  hasAnyData: boolean;
+  isAllRequestsComplete: boolean;
+  onRefreshTimeRange: () => void;
+  forceUpdate: string;
+}
+
+export const HasDataContext = createContext({} as HasDataContextValue);
+
+const apps: DataContextApps[] = ['apm', 'uptime', 'infra_logs', 'infra_metrics', 'ux', 'alert'];
+
+export function HasDataContextProvider({ children }: { children: React.ReactNode }) {
+  const { core } = usePluginContext();
+  const [forceUpdate, setForceUpdate] = useState('');
+  const { absStart, absEnd } = useTimeRange();
+
+  const [hasData, setHasData] = useState<HasDataContextValue['hasData']>({});
+
+  useEffect(
+    () => {
+      apps.forEach(async (app) => {
+        try {
+          if (app !== 'alert') {
+            const params =
+              app === 'ux' ? { absoluteTime: { start: absStart, end: absEnd } } : undefined;
```

Comment:
Why the special handling for alert and ux? I thought we had the same interface for all apps

Reply:
For `ux` there's a tread where @dgieselaar and @shahzad31 discussed it, https://github.com/elastic/kibana/pull/80644#discussion_r506273325. To remove it is necessary some refactor on the UX side, which would be better if we could fo it in another PR.

Reply:
For `alert`, I can solve it if I manually register the alert service with `registerDataHandler`. WDYT?